unityfiles/Assets/Scripts/Helper/Util/Timers.cs

```diff
@@ -0,0 +1,158 @@
+    /// <returns>true if a timer of the given id exists false if not</returns>
+    public bool Exists(string timerID){
+        return timers.Keys.Contains(timerID);
+    }
+
+    /// <summary>
+    /// returns the time left in the timer of the given id
+    /// if no timer has the id -1 is returned 
+    /// if timer has passed 0 is returned
+    /// </summary>
+    /// <param name="timerID">the timer to get time left from</param>
+    /// <returns>the time left in millisec</returns>
+    public int TimeLeft(string timerID){
+        int ret = -1;
+        if (timers.Keys.Contains(timerID)){
+            ret = this.timers[timerID].TimeLeft();
+            if (ret < 0){
+                ret = 0;
+            }
+        } 
+        return ret;
+    }
+
+    // -- overloads -- //
+
+    /// <summary>
+    /// Checks if a timer is complete returns true if it is false if not
+    /// if the timer is complete and restart is true the timer is restarted
+    /// if the timer is complete and restart is false the timer is removed
+    /// if no timer of the given id exists false is returned
+    /// </summary>
+    /// <param name="timerID">the id of the timer to check</param>
+    /// <param name="restart">If the timer should be restarted</param>
+    /// <returns>true if timer is done false if not</returns>
+    public bool Done(string timerID, bool restart) => this.isTimerDone(timerID, restart);
+
+    /// <summary>
+    /// Checks if a timer is complete returns true if it is false if not
+    /// if the timer is complete the timer is removed
+    /// if no timer of the given id exists false is returned
+    /// </summary>
+    /// <param name="timerID">the id of the timer to check</param>
+    /// <returns>true if timer is done false if not</returns>
+    public bool Done(string timerID) => this.isTimerDone(timerID, false);
+
+    // -- private -- // 
+
+    /// <summary>
+    /// Checks if a timer is complete returns true if it is false if not
+    /// if the timer is complete and restart is true the timer is restarted
+    /// if the timer is complete and restart is false the timer is removed
+    /// if no timer of the given id exists false is returned
+    /// </summary>
+    /// <param name="timerID">the id of the timer to check</param>
+    /// <param name="restart">If the timer should be restarted</param>
+    /// <returns>true if timer is done false if not</returns>
+    private bool isTimerDone(string timerID, bool restart)
+    {
+        bool retObj = false;
+        if (timers.ContainsKey(timerID))
+        {
+            Timer t = timers[timerID];
+            if (t.done())
+            {
+                if (restart)
+                {
+                    t.Restart();
+                } else {
+                    timers.Remove(timerID);
+                }
+                retObj = true;
+            } else
+            {
+                retObj = false;
+            }
+        }
+        return retObj;
+    }
+}
```

Comment:
change the name of "t" to something more readable

Reply:
nope this is a 10 line temporary variable